### PR TITLE
Fixes #10019: send-clean.sh blocks on inventory parsing error

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/10019

### CODE REVIEW

File: tools/send-clean.sh [hunk 1]
```diff
@@ -85,21 +85,36 @@ fi
 HTTP_CODE=$(${CURL_BINARY} --proxy '' -f ${SIGNATURE_OPT} -F "file=@${FILENAME}" -o /dev/null -w '%{http_code}' ${SERVER})
 SEND_COMMAND_RET=$?
 
-# 5 - Abort if sending failed
+# 5.1 - Wait is queue is overloaded
```
Comment (amousset) [inline tools/send-clean.sh]: wait if

File: tools/send-clean.sh [hunk 2]
```diff
@@ -85,21 +85,36 @@ fi
...
+if [ ${SEND_COMMAND_RET} -eq 7 -o "${HTTP_CODE}" -eq 503 ]
+then
+  # Endpoint is too busy (HTTP_CODE == 503), wait before trying other inventories
+  echo "WARNING: queue is overloaded, next inventories will be sent during next round"
```
Comment (amousset) [inline tools/send-clean.sh]: the message should be after the else

**Comment by peckpeck** [comment]
Commit modified

File: tools/send-clean.sh [hunk 3]
```diff
@@ -85,21 +85,36 @@ fi
 HTTP_CODE=$(${CURL_BINARY} --proxy '' -f ${SIGNATURE_OPT} -F "file=@${FILENAME}" -o /dev/null -w '%{http_code}' ${SERVER})
 SEND_COMMAND_RET=$?
 
-# 5 - Abort if sending failed
+# 5.1 - Wait if queue is overloaded
+if [ ${SEND_COMMAND_RET} -eq 7 -o "${HTTP_CODE}" -eq 503 ]
+then
+  # Endpoint is too busy (HTTP_CODE == 503), wait before trying other inventories
+  if [ "${sleeped_once}" = "" ]
```
Comment (amousset) [inline tools/send-clean.sh]: I don't see any loop here, how does this work?

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
OK, merging this PR

**Pull request merged**